pkg/kgo/source.go

```diff
@@ -344,8 +344,38 @@ func (s *source) hook(f *Fetch, buffered, polled bool) {
 }
 
 // takeBuffered drains a buffered fetch and updates offsets.
-func (s *source) takeBuffered() Fetch {
-	return s.takeBufferedFn(true, usedOffsets.finishUsingAllWithSet)
+func (s *source) takeBuffered(paused pausedTopics) Fetch {
+	var strip mtmps
```

Comment:
what do you think about returning early if `paused == nil`? It remove that after of the `eachOffset` "loop", e.g.
``` 
if paused == nil {
    return s.takeBufferedFn(true, usedOffsets.finishUsingAllWithSet)
}

var strip mtmps
f := s.takeBufferedFn(true, func(os usedOffsets) {
		os.eachOffset(func(o *cursorOffsetNext) {
			if paused.has(o.from.topic, o.from.partition) {
				o.from.allowUsable()
				strip.add(o.from.topic, o.from.partition)
				return
			}
			o.from.setOffset(o.cursorOffset)
			o.from.allowUsable()
		})
	})

Reply:
Need to check len 0 though rather than nil since paused is never entirely nilled out once it is initialized; this commit has been amended.